themes/githelpers.theme.bash

```diff
@@ -101,29 +117,51 @@ function _git-status-counts {
 }
 
 function _git-remote-info {
-  [[ "$(_git-upstream)" == "" ]] && return || true
-
-  [[ "$(_git-branch)" == "$(_git-upstream-branch)" ]] && local same_branch_name=true || true
-  local same_branch_name=
-  [[ "$(_git-branch)" == "$(_git-upstream-branch)" ]] && same_branch_name=true
-  if ([[ "${SCM_GIT_SHOW_REMOTE_INFO}" = "auto" ]] && [[ "$(_git-num-remotes)" -ge 2 ]]) ||
-      [[ "${SCM_GIT_SHOW_REMOTE_INFO}" = "true" ]]; then
-    if [[ "${same_branch_name}" != "true" ]]; then
-      remote_info="\$(_git-upstream)"
-    else
-      remote_info="$(_git-upstream-remote)"
+
+  if [[ "${SCM_GIT_GITSTATUSD_RAN}" == "true" ]]; then # prompt handling only, reimplement because patching the routine below gets ugly
+    [[ "${VCS_STATUS_REMOTE_NAME}" == "" ]] && return || true
+    [[ "${VCS_STATUS_LOCAL_BRANCH}" == "${VCS_STATUS_REMOTE_BRANCH}" ]] && local same_branch_name=true || true
+    local same_branch_name=
+    [[ "${VCS_STATUS_LOCAL_BRANCH}" == "${VCS_STATUS_REMOTE_BRANCH}" ]] && same_branch_name=true
+    if [[ "${SCM_GIT_SHOW_REMOTE_INFO}" = "true" ]] || [[ "${SCM_GIT_SHOW_REMOTE_INFO}" = "auto" ]]; then # no multiple remote support in gitstatusd
+      if [[ "${same_branch_name}" != "true" ]]; then
+        remote_info="\${VCS_STATUS_REMOTE_NAME}"
+      else
+        remote_info="${VCS_STATUS_REMOTE_NAME}/${VCS_STATUS_REMOTE_BRANCH}"
+      fi
+    elif [[ ${same_branch_name} != "true" ]]; then
+      remote_info="\${VCS_STATUS_REMOTE_BRANCH}"
     fi
-  elif [[ ${same_branch_name} != "true" ]]; then
-    remote_info="\$(_git-upstream-branch)"
-  fi
-  if [[ -n "${remote_info}" ]];then
-    local branch_prefix
-    if _git-upstream-branch-gone; then
-      branch_prefix="${SCM_THEME_BRANCH_GONE_PREFIX}"
-    else
-      branch_prefix="${SCM_THEME_BRANCH_TRACK_PREFIX}"
+    if [[ -n "${remote_info}" ]];then
+      local branch_prefix 
+      branch_prefix="${SCM_THEME_BRANCH_TRACK_PREFIX}" # no support for gone remote branches in gitstatusd
+      echo "${branch_prefix}${remote_info}"
+    fi
+  else 
+    [[ "$(_git-upstream)" == "" ]] && return || true
```

Comment:
I don't know bash very well. Will this result in a fork?

Reply:
That is a good question. I do not know, I'm not a bash wizard either.
But this whole remote info code is very ugly and inefficient anyway, that is why I basically reimplemented it completely using the VCS variables instead of patching the functions like I did anywhere else.
Someone who is more into bash and output parsing should refactor this in another PR.
For now it works ok.